src/hwdb/trie.rs

```diff
@@ -25,8 +25,8 @@ pub fn trie_string(hwdb_buf: &[u8], offset: usize) -> &str {
             .map(|end| offset + end)
             .unwrap_or(buf_len);
 
-        std::str::from_utf8(&hwdb_buf[offset..str_end]).unwrap_or("")
+        std::str::from_utf8(&hwdb_buf[offset..str_end]).expect("invalid UTF-8 string")
```

Comment:
If we're going to error on failing to create a `&str` we should change the return to a `Result` type, and bubble-up the error. I would prefer to remove as many `panic`s as possible.

Reply:
Yes you're right. I've done the work to change this to a `Result` return instead of "". From my understanding this conditions are not normal and it should propagate rather than silently return "".